### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by YuriyBilskiy**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
```
Comment (loralevitska) [inline src/transformState.js:8]: Destructure values from `action`

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,29 @@
...
+
+      for (const key of keysToRemove) {
+        delete state[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
```
Comment (loralevitska) [inline src/transformState.js]: Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others

**Review by loralevitska — CHANGES_REQUESTED** [review]
Almost done) Let's improve

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
```
Comment (sTorba24) [inline src/transformState.js:9]: you can do it directly in loop like 

const { type, extraData, keysToRemove } of actions

**Review by sTorba24 — APPROVED** [review]
GJ 🚀

**Pull request closed without merging**